modules/core/shared/src/main/scala/neotype/common/package.scala

```diff
@@ -0,0 +1,8 @@
+package neotype.common
+
+import neotype.*
+
+type NonEmptyString = NonEmptyString.Type
+object NonEmptyString extends Subtype[String]:
+  override inline def validate(input: String): Boolean =
+    input.nonEmpty
```

Comment:
It might be nice to have a custom error here. 
So

```scala
  override inline def validate(input: String): Boolean | String =
     if input.nonEmpty then true else "NonEmptyString cannot be empty"
```